freqtrade/freqai/freqai_interface.py

```diff
@@ -738,6 +744,29 @@ def _set_train_queue(self):
                     f'Best approximation queue: {best_queue}')
         return best_queue
 
+    def cache_corr_pairlist_dfs(self, dataframe: DataFrame, dk: FreqaiDataKitchen) -> DataFrame:
+        """
+        Cache the corr_pairlist dfs to speed up performance for subsequent pairs during the
+        current candle.
+        :param dataframe: strategy fed dataframe
+        :param dk: datakitchen object for current asset
+        :return: dataframe to attach/extract cached corr_pair dfs to/from.
+        """
+
+        if self.get_corr_dataframes:
+            self.corr_dataframes = dk.extract_corr_pair_columns_from_populated_indicators(dataframe)
+            if not self.corr_dataframes:
+                logger.warning("Couldn't cache corr_pair dataframes for improved performance. "
+                               "Consider ensuring that the full coin/stake, e.g. XYZ/USD, "
+                               "is included in the column names when you are creating features "
+                               "in `populate_any_indicators()`.")
+            self.get_corr_dataframes = not bool(self.corr_dataframes)
```

Comment:
I'm not sure i understand the logic used for `get_corr_dataframes` (when is it expected to be True, when false?)

from my understanding (please correct me where/if i've got it wrong):
* it's True at the beginning of each loop (it's set in "inference.stop" - so it'll be empty afterwards)
* it's it's False if we were able to extract columns from a populated DF (this line)
* it's kept at True if the above was successful
* if it's False - we use cached columns (?) -> else branch in this function

in my understanding, this won't be used by the training loop? do we have to somehow ensure that's not regressing in the future somehow - by adding some additional new functionality or whatnot ... ?

Reply:
If `self.corr_dataframes` is empty, then it will keep `self.get_corr_dataframes` as true in `use_strategy_to_populate_indicators` (`do_corr_pairs = True`). This means that it will always calculate the indicators for the corr_pair dfs for each main pair. This is the behavior of current develop. 

If `self.corr_dataframes` is not empty, it means that `extract_corr_pair_columns_from_populated_idicators` successfully found the corr_pairs in the column names. So it sets `self.get_corr_dataframes` to False, because, it just got them and cached them. All subsequent pairs during the present candle will `use_strategy_to_populate_indicators` with `do_corr_pairs = False` (it wont calculate the indicators).

At the end of the candle, `inference_timer` will set `self.get_corr_dataframes` to true and that means that the first pair to get inferenced during the next candle will calculate the full set of indicators, cache them, and then the cycle continues.

None of this has anything to do with training. This is only for inference.